crypto/x509/v3_sxnet.c

```diff
@@ -172,7 +172,7 @@ int SXNET_add_id_INTEGER(SXNET **psx, ASN1_INTEGER *zone, const char *user,
         sx = *psx;
     if (SXNET_get_id_INTEGER(sx, zone)) {
         ERR_raise(ERR_LIB_X509V3, X509V3_R_DUPLICATE_ZONE_ID);
-        return 0;
+        goto err;
```

Comment:
This is wrong here as there is `ERR_raise(ERR_LIB_X509V3, ERR_R_MALLOC_FAILURE);` in the err path, which should not be called in this case. Also it would be probably wrong to deallocate the *psx in case it was not allocated by the call.